jax/api.py

```diff
@@ -1780,7 +1784,11 @@ def fun(*tangents):
 
 def _check_inexact_input_vjp(x):
```

Comment:
Should we just get rid of this check entirely? On one hand it seems like we're expanding the scope of autodiff to handle exact dtypes; on the other hand it might catch some accidental errors if people accidentally write `grad(f)(0)` when they mean `grad(f)(0.)`.

My current hunch is we should remove this check.

Reply:
I think it's a common enough mistake to use integer arguments where you intended floats, and it now errors clearly while after this change it will return these `float0`s which require some (non-trivial) context.
Ideally there would be an error on any unsupported operation on the `Float0Array` explaining you differentiated wrt an integer argument. I like the idea of adding the source line of where the array was created (I think I've seen such errors already in context of omnistaging, so would be cool to replicate that)

Is there a way to override all errors when trying to use an unimplemented method or should I add it for the most commonly used ones?